mutate/mutate.go

```diff
@@ -291,21 +272,20 @@ func (m *Mutator) add(ctx context.Context, reader io.Reader, history *ispec.Hist
 // generate the DiffIDs for the image metatadata. The provided history entry is
 // appended to the image's history and should correspond to what operations
 // were made to the configuration.
-func (m *Mutator) Add(ctx context.Context, r io.Reader, history *ispec.History) (ispec.Descriptor, error) {
+func (m *Mutator) Add(ctx context.Context, r io.Reader, history *ispec.History, compressor Compressor) (ispec.Descriptor, error) {
 	desc := ispec.Descriptor{}
 	if err := m.cache(ctx); err != nil {
 		return desc, errors.Wrap(err, "getting cache failed")
 	}
 
-	digest, size, err := m.add(ctx, r, history)
+	digest, size, err := m.add(ctx, r, history, compressor)
 	if err != nil {
 		return desc, errors.Wrap(err, "add layer")
 	}
 
 	// Append to layers.
 	desc = ispec.Descriptor{
-		// TODO: Detect whether the layer is gzip'd or not...
-		MediaType: ispec.MediaTypeImageLayerGzip,
+		MediaType: compressor.MediaType(),
```

Comment:
As I said in the other comment this should be something like *shudder* `ispec.MediaTypeImageLayer + compressor.MediaType()`. I get that you folks don't use layers `.tar.gz` layers -- but that means we should extend `Mutator` to separately support using a custom media-type IMHO. Custom compression is kind of separate to custom layer media-types.